bokeh/models/map_plots.py

```diff
@@ -36,6 +36,21 @@ class MapPlot(Plot):
 
     '''
 
+    def __init__(self, *args, **kw):
+        from ..models.ranges import Range1d
+        for r in ('x_range', 'y_range'):
+            if not isinstance(kw.get(r, None), Range1d):
+                raise ValueError('Invalid value for %r, MapPlot ranges may only be Range1d, not data ranges' % r)
```

Comment:
Why is this necessary? I would expect this should be handled solely by validation.

Reply:
I presume this is because validation happens a bit late.

Reply:
It happens late, and also we have not (at least not yet) made validation errors raise real exceptions (they just output the message and proceed)